internal/analytics/analytics.go

```diff
@@ -0,0 +1,101 @@
+package analytics
+
+import (
+	ph "github.com/posthog/posthog-go"
+
+	"github.com/datumforge/datum/internal/analytics/posthog"
+)
+
+var _ Handler = (*posthog.PostHog)(nil)
+
+var (
+	handler Handler
+)
+
+func init() {
+	p := posthog.Init()
+	if p != nil {
+		handler = p
+	}
+}
+
+// EventManager isn't your normal party planner
+type EventManager struct {
+	EM Handler
```

Comment:
nit: 

```suggestion
	Handler Handler
```

This should be called Handler, otherwise you have a "stutter" "EventManager.EM" when you call it